### PR TITLE
Some more fix for PR#33582

### PR DESCRIPTION
**Pull request by gpei**
As the debugging on https://github.com/openshift/release/pull/33582 is not finished yet but PR got merged unexpectedly, so continue the follow-up fix and enhancement here.

### CODE REVIEW

File: ci-operator/step-registry/workers-rhel/azure-provision/workers-rhel-azure-provision-commands.sh [hunk 1]
```diff
@@ -37,19 +37,19 @@ fi
...
-  # check if the current region support AZ by checking if $az_num != 0
-  IFS=$'\t' read -r -a zone_list <<< "$(az vm list-skus -l ${AZURE_REGION} --zone --size ${RHEL_VM_SIZE} --query '[].locationInfo[0].zones' -o tsv)"
-  az_num=${#zone_list[@]}
-
   # az command to create RHEL VM and append --zone when the region has AZ
   cmd="az vm create --resource-group '${infra_id}-rg' --name '${infra_id}-rhel-${count}' --image '${RHEL_IMAGE}' --ssh-key-values '${SSH_PUB_KEY_PATH}' --admin-user '${RHEL_USER}' --public-ip-address '' --size '${RHEL_VM_SIZE}' --os-disk-size-gb '${RHEL_VM_DISK_SIZE}' --nsg '' --subnet '${computeSubnetID}'"
 
   if [ "${az_num}" != "0" ]; then
-    cmd="${cmd} --zone ${count}"
+    cmd="${cmd} --zone $((${count} % ${az_num} + 1))"
```
Comment (MayXuQQ) [inline ci-operator/step-registry/workers-rhel/azure-provision/workers-rhel-azure-provision-commands.sh:52]: cmd="${cmd} --zone $((${count} % ${az_num} ))" ?

Comment (gpei) [reply ci-operator/step-registry/workers-rhel/azure-provision/workers-rhel-azure-provision-commands.sh:52]: For zone name, it's always in the collection of [1,2,3] right? we shouldn't make the zone name as "0"

Comment (MayXuQQ) [reply ci-operator/step-registry/workers-rhel/azure-provision/workers-rhel-azure-provision-commands.sh:52]: got it

**Comment by gpei** [comment]
In the test job https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/33623/rehearse-33623-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.11-azure-ipi-workers-rhel8-p2/1587318585431166976

we can see the RHEL machine is provisioned as expected
```
2022-11-01 08:06:48+00:00 - Provision ci-op-cypr8b8j-cd84e-l7bpq-rhel-1 ...
Creating RHEL VM: az vm create --resource-group 'ci-op-cypr8b8j-cd84e-l7bpq-rg' --name 'ci-op-cypr8b8j-cd84e-l7bpq-rhel-1' --image 'RedHat:RHEL:8-lvm-gen2:8.6.2022051814' --ssh-key-values '/var/run/secrets/ci.openshift.io/cluster-profile/ssh-publickey' --admin-user 'cloud-user' --public-ip-address '' --size 'Standard_D4s_v3' --os-disk-size-gb '64' --nsg '' --subnet '/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-cypr8b8j-cd84e-rg/providers/Microsoft.Network/virtualNetworks/ci-op-cypr8b8j-cd84e-vnet/subnets/ci-op-cypr8b8j-cd84e-worker-subnet' --zone 2
{
  "fqdns": "",
  "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-cypr8b8j-cd84e-l7bpq-rg/providers/Microsoft.Compute/virtualMachines/ci-op-cypr8b8j-cd84e-l7bpq-rhel-1",
  "location": "centralus",
  "macAddress": "00-22-48-46-83-80",
  "powerState": "VM running",
  "privateIpAddress": "10.0.1.7",
  "publicIpAddress": "",
  "resourceGroup": "ci-op-cypr8b8j-cd84e-l7bpq-rg",
  "zones": "2"
}
```
and RHCOS worker could be drained successfully now.
```
+ echo '2022-11-01 08:25:51+00:00 - Waiting for CoreOS nodes to be removed'
2022-11-01 08:25:51+00:00 - Waiting for CoreOS nodes to be removed
+ oc wait node --for=delete --timeout=10m --selector node.openshift.io/os_id=rhcos,node-role.kubernetes.io/worker
node/ci-op-cypr8b8j-cd84e-l7bpq-worker-centralus1-ldc92 condition met
node/ci-op-cypr8b8j-cd84e-l7bpq-worker-centralus2-6dqlh condition met
node/ci-op-cypr8b8j-cd84e-l7bpq-worker-centralus3-lp4dv condition met
++ date -u --rfc-3339=seconds
+ echo '2022-11-01 08:28:58+00:00 - Waiting for worker machineconfigpool to update'
2022-11-01 08:28:58+00:00 - Waiting for worker machineconfigpool to update
+ oc wait machineconfigpool/worker --for=condition=Updated=True --timeout=10m
machineconfigpool.machineconfiguration.openshift.io/worker condition met
++ date -u --rfc-3339=seconds
+ echo '2022-11-01 08:28:59+00:00 - Waiting for clusteroperators to complete'
2022-11-01 08:28:59+00:00 - Waiting for clusteroperators to complete
+ oc wait clusteroperator.config.openshift.io --for=condition=Available=True --for=condition=Progressing=False --for=condition=Degraded=False --timeout=20m --all
clusteroperator.config.openshift.io/authentication condition met
clusteroperator.config.openshift.io/baremetal condition met
clusteroperator.config.openshift.io/cloud-controller-manager condition met
clusteroperator.config.openshift.io/cloud-credential condition met
clusteroperator.config.openshift.io/cluster-autoscaler condition met
clusteroperator.config.openshift.io/config-operator condition met
clusteroperator.config.openshift.io/console condition met
clusteroperator.config.openshift.io/csi-snapshot-controller condition met
clusteroperator.config.openshift.io/dns condition met
clusteroperator.config.openshift.io/etcd condition met
clusteroperator.config.openshift.io/image-registry condition met
clusteroperator.config.openshift.io/ingress condition met
clusteroperator.config.openshift.io/insights condition met
clusteroperator.config.openshift.io/kube-apiserver condition met
clusteroperator.config.openshift.io/kube-controller-manager condition met
clusteroperator.config.openshift.io/kube-scheduler condition met
clusteroperator.config.openshift.io/kube-storage-version-migrator condition met
clusteroperator.config.openshift.io/machine-api condition met
clusteroperator.config.openshift.io/machine-approver condition met
clusteroperator.config.openshift.io/machine-config condition met
clusteroperator.config.openshift.io/marketplace condition met
clusteroperator.config.openshift.io/monitoring condition met
clusteroperator.config.openshift.io/network condition met
clusteroperator.config.openshift.io/node-tuning condition met
clusteroperator.config.openshift.io/openshift-apiserver condition met
clusteroperator.config.openshift.io/openshift-controller-manager condition met
clusteroperator.config.openshift.io/openshift-samples condition met
clusteroperator.config.openshift.io/operator-lifecycle-manager condition met
clusteroperator.config.openshift.io/operator-lifecycle-manager-catalog condition met
clusteroperator.config.openshift.io/operator-lifecycle-manager-packageserver condition met
clusteroperator.config.openshift.io/service-ca condition met
clusteroperator.config.openshift.io/storage condition met
++ date -u --rfc-3339=seconds
+ echo '2022-11-01 08:29:04+00:00 - RHEL worker scaleup complete'
2022-11-01 08:29:04+00:00 - RHEL worker scaleup complete
```
I believe this is ready to be merged now.

**Comment by gpei** [comment]
@jianlinliu @jinyunma this is ready for review now, I also updated the `RHEL_IMAGE` in all Azure RHEL jobs to 8.6 in this PR as 8.6 is the only one we'll need to test for 4.12 based on latest Test Matrix. 

BTW, @jinyunma those RHEL images are published by Azure cloud officially, not like vSphere where we have to upload image by ourselves, command `az vm image list --offer RHEL --publisher RedHat --all --sku 8-lvm-gen2` could list the available RHEL-8 images, so when new RHEL-8.7 image published, we can also get its image name from the list.

**Comment by jianlinliu** [comment]
@gpei can you help me recall the reason of setting the default workers number to `3`?

**Comment by gpei** [comment]
> @gpei can you help me recall the reason of setting the default workers number to `3`?

Sure, for a normal IPI Azure cluster installed in AZ available region, it will have 3 RHCOS created in the 3 different zones. 
And some pvc were created with specific zone matching rules, such as:

```
$ oc get pv pvc-3d7251d5-6167-4245-8f12-effa11a6bcc8 -ojson | jq .spec.nodeAffinity
{
  "required": {
    "nodeSelectorTerms": [
      {
        "matchExpressions": [
          {
            "key": "topology.disk.csi.azure.com/zone",
            "operator": "In",
            "values": [
              "westus3-1"
            ]
          }
        ]
      }
    ]
  }
}
```

So when the pod using this PV got re-scheduled to another node and mount to the same volume, the node has to be in the same zone.

For this workflow `cucushift-installer-rehearse-azure-ipi-workers-rhel8-provision`, it will only has RHEL-8 workers in the end. We also need at least 3 RHEL workers created in the 3 different zones, or else pods couldn't be rescheduled, RHCOS worker couldn't be drained and removed.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/proxy/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-azure-ipi-proxy-workers-rhcos-rhel8-provision-chain.yaml [hunk 2]
```diff
@@ -17,10 +17,10 @@ chain:
     default: "Standard_D4s_v3"
     documentation: The Size of the RHEL VM provisioned.
   - name: RHEL_IMAGE
-    default: "RedHat:RHEL:8-lvm-gen2:8.5.2021121504"
+    default: "RedHat:RHEL:8-lvm-gen2:8.6.2022051814"
```
Comment (MayXuQQ) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/proxy/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-azure-ipi-proxy-workers-rhcos-rhel8-provision-chain.yaml:20]: RedHat:RHEL:8-lvm-gen2:8.6.2022102701

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/workers-rhel8/provision/cucushift-installer-rehearse-azure-ipi-workers-rhel8-provision-chain.yaml [hunk 3]
```diff
@@ -16,13 +16,13 @@ chain:
   - chain: workers-rhel
   env:
   - name: RHEL_WORKER_COUNT
-    default: 2
+    default: 3
   - name: RHEL_VM_SIZE
     default: "Standard_D4s_v3"
   - name: RHEL_IMAGE
-    default: "RedHat:RHEL:8-lvm-gen2:8.5.2021121504"
+    default: "RedHat:RHEL:8-lvm-gen2:8.6.2022051814"
```
Comment (MayXuQQ) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/workers-rhel8/provision/cucushift-installer-rehearse-azure-ipi-workers-rhel8-provision-chain.yaml:23]: RedHat:RHEL:8-lvm-gen2:8.6.2022102701

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/proxy/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-azure-ipi-proxy-workers-rhcos-rhel8-provision-chain.yaml (hunk 2 above)
Comment (gpei) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/proxy/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-azure-ipi-proxy-workers-rhcos-rhel8-provision-chain.yaml:20]: I guess this is fine as they're actually the same RHEL major version.

**Pull request merged**